lib/src/file_checksum.dart

```diff
@@ -4,29 +4,52 @@ import 'dart:io';
 import 'package:crypto/crypto.dart';
 import 'package:crypto/src/digest_sink.dart';
 
+import 'exceptions.dart';
+
 class FileChecksum {
   final Stream<List<int>> fileContents;
   Digest md5Digest;
 
   static Future<String> getMd5AsBase64({File file}) async {
-    var checksum = FileChecksum(file.openRead());
+    Stream<List<int>> _fileContents;
+    try {
+      _fileContents = file.openRead();
+    } on UnsupportedError catch (e) {
+      throw FileChecksumException(e.message);
+    } on FileSystemException catch (e) {
+      throw FileChecksumException(e.message);
+    }
+    FileChecksum checksum = FileChecksum(_fileContents);
+    ;
     await checksum.compute();
     return checksum.base64;
   }
 
   FileChecksum(this.fileContents);
 
-  String get base64 => base64Encode(rawBytes);
+  String get base64 {
+    try {
+      return base64Encode(rawBytes);
+    } on FormatException catch (e) {
+      throw FileChecksumException(e.message);
+    } on ArgumentError catch (e) {
+      throw FileChecksumException(e.message);
+    }
+  }
```

Comment:
@safeforge Unfortunately, the `on` keywork only accepts 1 type argument. Also, I only want to catch `FormatException` and `ArgumentError`; other exceptions should not be caught. If you have something specific in mind please share a snippet 🙂

Reply:
I think `base64Encode` can raises only `FormatException` or `ArgumentError` and the code could be simplify with

```dart
try {
  return base64Encode(rawBytes);
} catch (e) {
  throw FileChecksumException(e.message);
}
```

Reply:
> I think `base64Encode` can raises only `FormatException` or `ArgumentError`

I'm totally aware of that, I inspected the source code to determine the exceptions to catch 😉 

I agree that my code has some repetition, but I think it's safer. After some Dart update, `base64Encode` might throw new types of exceptions, maybe custom ones that won't have the `message` attribute (the `Exception` interface doesn't have `message`). In such case, I'd prefer the code to crash with the original exception, rather than with a generic "method not found" 😉

Reply:
Thank you for great explanation.